src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -636,8 +652,18 @@ def delete_connectedk8s(cmd, client, resource_group_name, cluster_name,
                                            error_message="Unable to read ConfigMap 'azure-clusterconfig' in 'azure-arc' namespace: ",
                                            message_for_not_found="The helm release 'azure-arc' is present but the azure-arc namespace/configmap is missing. Please run 'helm delete azure-arc --no-hooks' to cleanup the release before onboarding the cluster again.")
 
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
     if (configmap.data["AZURE_RESOURCE_GROUP"].lower() == resource_group_name.lower() and
-            configmap.data["AZURE_RESOURCE_NAME"].lower() == cluster_name.lower()):
+            configmap.data["AZURE_RESOURCE_NAME"].lower() == cluster_name.lower() and configmap.data["AZURE_SUBSCRIPTION_ID"].lower() == subscription_id.lower()):
+
+        armid = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Kubernetes/connectedClusters/{}".format(subscription_id, resource_group_name, cluster_name)
+        arm_hash = hashlib.sha256(armid.lower().encode('utf-8')).hexdigest()
+
+        if check_proxy_kubeconfig(kube_config, kube_context, arm_hash):
+            telemetry.set_user_fault()
```

Comment:
add a exception type for telemetry purpose

Reply:
added